docs/apidoc/python/apigen.rst

```diff
@@ -0,0 +1,512 @@
+
+      python_apigen_order_tiebreaker = "alphabetical"
+
+.. confval:: python_apigen_case_insensitive_filesystem
+
+   This extension results in an output file for each documented Python object
+   based on its fully-qualified name.  Python names are case-sensitive, meaning
+   both :python:`foo` and :python:`Foo` can be defined within the same scope,
+   but some filesystems are case insensitive (e.g. on Windows and macOS), which
+   creates the potential for a conflict.
+
+   By default (if :confval:`python_apigen_case_insensitive_filesystem` is
+   :python:`None`), this extension detects automatically if the filesystem is
+   case-insensitive, but detection is skipped if the option is set to an
+   explicit value of :python:`True` or :python:`False`:
+
+   .. code-block:: python
+      :caption: Add to :file:`conf.py` to force case-insensitive naming scheme
+
+      python_apigen_case_insensitive_filesystem = True
+
+   If the filesystem is either detected or specified to be case-insensitive,
+   case conflicts are avoided by including a hash in the document name.
+
+Subscript methods
+^^^^^^^^^^^^^^^^^
+
+*Subscript methods* are attributes defined on an object that support subscript
+syntax.  For example:
+
+.. code-block:: python
+
+   arr.vindex[1, 2:5, [1, 2, 3]]
+
+These subscript methods can be implemented as follows:
+
+.. code-block:: python
+
+   class MyArray:
+       class _Vindex:
+           def __init__(self, arr: MyArray):
+               self.arr = arr
+
+           def __getitem__(self, sel: Selection):
+               # Do something with `self.arr` and `sel`.
+               return result
+
+       @property
+       def vindex(self) -> MyArray._Vindex:
+           return MyArray._Vindex(self)
+
+Based on the :confval:`python_apigen_subscript_method_types` option, this
+extension can recognize this pattern and display :python:`vindex` as:
+
+.. code-block::
+
+   vindex[sel: Selection]
+
+rather than as a normal property.
+
+.. confval:: python_apigen_subscript_method_types
+
+   Regular expression pattern that matches the return type annotations of
+   properties that define subscript methods.
+
+   Return type annotations can be specified either as real annotations or in the
+   textual signature specified as the first line of the docstring.
+
+   The default value matches any name beginning with an underscore,
+   e.g. :python:`_Vindex` in the example above.
+
+.. confval:: python_apigen_show_base_classes
+
+   Display the list of base classes when documenting classes.
+
+   Unlike the built-in `sphinx.ext.autodoc` module, base classes are shown using
+   the normal Python syntax in a parenthesized list after the class name.
+
+   The list of base classes displayed for each class can be customized by adding
+   a listener to the `autodoc-process-bases` event.  This is useful for
+   excluding base classes that are not intended to be part of the public API.
```

Comment:
This is actually present in the Sphinx `objects.inv` file, but under the `std:event` object type.  Due to how intersphinx works, you can't reference an object type that isn't defined in the current documentation build.  Therefore, I added a definition for that object type to our `conf.py` so that we can reference events.